tests/unit/config/test_config.py

```diff
@@ -447,6 +447,30 @@ def test_master_pillar_roots_glob(self):
             if os.path.isdir(tempdir):
                 shutil.rmtree(tempdir)
 
+    def test_master_id_function(self):
+        try:
+            tempdir = tempfile.mkdtemp(dir=TMP)
+            master_config = os.path.join(tempdir, 'master')
```

Comment:
I know you've just copied this from other tests, but normally in the unit tests we are always mocking the file descriptors instead of writing actual files _nowadays_. Because this is not always true that your unit test env has a write access to the temporary system directory, it can be a limited resources sandbox in a read-only docker, for example. Unit tests supposed to run everywhere without touching actual things. Integration tests actually do writing data. Therefore please mock this one instead.